CMake/FindMKL.cmake

```diff
@@ -2,42 +2,106 @@
 # This needs a lot of work to make it robust
 INCLUDE( CheckCXXSourceCompiles )
 
+if ( NOT CMAKE_CXX_COMPILER_ID MATCHES "Intel" )
+  # if not intel and MKL_ROOT not set
+  if (NOT MKL_ROOT)
+    find_path(MKL_ROOT "mkl.h"
+      HINTS ${MKLROOT} ${MKL_HOME} $ENV{MKLROOT} $ENV{MKL_ROOT} $ENV{MKL_HOME}
+      PATH_SUFFIXES include)
+    string(REPLACE "/include" "" MKL_ROOT ${MKL_ROOT})
+  endif (NOT MKL_ROOT)
+  if (NOT MKL_ROOT)
+    message (FATAL_ERROR "ENABLE_OUTSIDE_MKL is TRUE and mkl not found. Set MKL_ROOT." )
+  endif (NOT MKL_ROOT)
+
+  #Do we support ilp?
+  set(MKL_FIND_LIB "libmkl_intel_lp64${CMAKE_SHARED_LIBRARY_SUFFIX}")
+  set(SUFFIXES lib lib/intel64)
+  find_path(MKL_LINK_DIRECTORIES name "${MKL_FIND_LIB}" HINTS ${MKL_ROOT}
+    PATH_SUFFIXES ${SUFFIXES})
+  if (MKL_LINK_DIRECTORIES-NOTFOUND)
+    message(FATAL_ERROR "MKL_LINK_DIRECTORIES not set. ${MKL_FIND_LIB} "
+      "not found in MKL_ROOT/(${SUFFIXES})")
+  endif (MKL_LINK_DIRECTORIES-NOTFOUND)
+  message("MKL_LINK_DIRECTORIES: ${MKL_LINK_DIRECTORIES}")
+  set(SUFFIXES include)
+  find_path(MKL_INCLUDE_DIRECTORIES name "mkl.h" HINTS ${MKL_ROOT}
+    PATH_SUFFIXES ${SUFFIXES})
+  if (MKL_INCLUDE_DIRECTORIES-NOTFOUND)
+    message(FATAL_ERROR "MKL_INCLUDE_DIRECTORIES not set. \"mkl.h\" not found in MKL_ROOT/(${SUFFIXES})")
+  endif (MKL_INCLUDE_DIRECTORIES-NOTFOUND)
+  message("MKL_INCLUDE_DIRECTORIES: ${MKL_INCLUDE_DIRECTORIES}")
+  set(MKL_FLAGS "-m64")
```

Comment:
What is this flag? it seems compiler specific not MKL related.

Reply:
Actually this is overwritten two lines later.

Reply:
The overwrite is a mistake but it should really be in clang compilers.

Reply:
After this PR, FindMKL should be general for GNU, Clang, PGI...
I don't see a reason to add this. From Intel linker advisor, there is no -m64 listed.

Reply:
So I or anyone else can use llvm 5 Abstract Syntax Tree based tools like rtags. Over time it could expand into full featured not intel compilers and MKL support but it doesn't need to for my goal to be accomplished.  It's not a full "feature" in one PR it is something I'm implementing so I can do more important work.

Reply:
At the moment, you can always pass it in via CMAKE_CXX_FLAGS.